subworkflows/local/sourmash_profiling.nf

```diff
@@ -25,16 +43,42 @@ workflow SOURMASH_PROFILING {
                 meta.id = "k31"
                 [ meta, signatures ]
         }
+
     SOURMASH_COMPARE(ch_compare, seqtype, [], true, true)
     ch_compare_matrix = SOURMASH_COMPARE.out.matrix
     ch_compare_csv = SOURMASH_COMPARE.out.csv
     ch_versions = ch_versions.mix(SOURMASH_COMPARE.out.versions)
 
     // gather against database
+    // prep all combinations of input files with combined databases
+    ch_input_gather = ch_signatures
+        .combine(ch_sourmash_databases)
+
+    SOURMASH_GATHER(ch_input_gather, seqtype,
+        [], // val save_unassigned
+        [], // val save_matches_sig
+        [], // val save_prefetch
+        []  // val save_prefetch_csv
+    )
+    ch_gather_result = SOURMASH_GATHER.out.result
+    ch_versions = ch_versions.mix(SOURMASH_GATHER.out.versions)
+
+    // taxonomy against lineage CSV
+    // combine per sample all lineage CSVs
+    ch_input_taxannotate = ch_gather_result
```

Comment:
same comment as above

Reply:
same logic so that all the databases/taxonomy options that are input in the sourmash_dbs CSV are run once per sample, which is what combine is doing once the `.collect()` and `.toList()` for the databases or lineage CSVs is set up and it's queuing one sample per list of databases/taxonomy CSVs.